airflow/www/security.py

```diff
@@ -347,19 +347,23 @@ def can_read_dag(self, dag_id, user=None) -> bool:
         if not user:
             user = g.user
         prefixed_dag_id = self.prefixed_dag_id(dag_id)
-        return self._has_view_access(
-            user, permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG
-        ) or self._has_view_access(user, permissions.ACTION_CAN_READ, prefixed_dag_id)
+        return (
+            self._has_view_access(user, permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG)
+            or self._has_view_access(user, permissions.ACTION_CAN_READ, prefixed_dag_id)
+            or False
+        )
 
     def can_edit_dag(self, dag_id, user=None) -> bool:
         """Determines whether a user has DAG edit access."""
         if not user:
             user = g.user
         prefixed_dag_id = self.prefixed_dag_id(dag_id)
 
-        return self._has_view_access(
-            user, permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG
-        ) or self._has_view_access(user, permissions.ACTION_CAN_EDIT, prefixed_dag_id)
+        return (
+            self._has_view_access(user, permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG)
+            or self._has_view_access(user, permissions.ACTION_CAN_EDIT, prefixed_dag_id)
+            or False
```

Comment:
with mssql, `self._has_view_access` seems to return `None` instead of `False` and hence it was failing with `TypeError: unsupported operand type(s) for |=: 'NoneType' and 'NoneType'`  at  https://github.com/apache/airflow/blob/6c9df7c658f664c8296af887a2796b3ac082649e/airflow/www/security.py#L406-L410

may be I should instead  change that to 

```
        if self.is_dag_resource(resource):
            if permission == permissions.ACTION_CAN_READ:
                has_access = has_access or self.can_read_dag(resource, user)
            elif permission == permissions.ACTION_CAN_EDIT:
                has_access = has_access or self.can_edit_dag(resource, user)
```  

?

Reply:
I have a PR that wraps `_has_view_access` (we're changing the naming from `view` and `permission` to `resource` and `action`). We can just bake sure the output of the `_has_access()` wrapping function returns a boolean.

https://github.com/apache/airflow/pull/15398/files#diff-9230f8648061da203f769faa144f992581fe146a271233b2874a522bd63544aaR441

Reply:
cool, will wait for it to be merged :)